go/vt/sqlparser/ast.go

```diff
@@ -1626,6 +1637,42 @@ type PartitionValueRange struct {
 	Maxvalue bool
 }
 
+type PartitionEngine struct {
+	Storage bool
+	Equal   string
+	Name    string
+}
+
+type PartitionComment struct {
+	Equal   string
```

Comment:
You can drop all the `Equal` fields. Having an `=` in the query where it is optional doesn't change the meaning of the query and is therefore of no use to store in the struct. You can look at the other places where we have used `equal_opt` in sql.y

Reply:
Fixed